src/components/alarm/AlarmController.cpp

```diff
@@ -81,8 +76,9 @@ void AlarmController::ScheduleAlarm() {
 
   // now can convert back to a time_point
   alarmTime = std::chrono::system_clock::from_time_t(std::mktime(tmAlarmTime));
-  auto mSecToAlarm = std::chrono::duration_cast<std::chrono::milliseconds>(alarmTime - now).count();
-  app_timer_start(alarmAppTimer, APP_TIMER_TICKS(mSecToAlarm), this);
+  auto secondsToAlarm = std::chrono::duration_cast<std::chrono::seconds>(alarmTime - now).count();
+  xTimerChangePeriod(alarmTimer, secondsToAlarm * configTICK_RATE_HZ, 0);
```

Comment:
In MotorController the `pdMS_TO_TICKS` macro is used, and here the `configTICK_RATE_HZ` define is used directly. I think we should use the same conversion method in both cases

```suggestion
  auto mSecToAlarm = std::chrono::duration_cast<std::chrono::milliseconds>(alarmTime - now).count();
  xTimerChangePeriod(alarmTimer, pdMS_TO_TICKS(mSecToAlarm), 0);
```

Reply:
This is intentional. See https://github.com/InfiniTimeOrg/InfiniTime/pull/1095#discussion_r871873957